sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/FieldBuilderOptions.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.indexes.models;
+
+import com.azure.core.util.serializer.MemberNameConverter;
+import com.azure.search.documents.indexes.SearchIndexAsyncClient;
+import com.azure.search.documents.indexes.SearchIndexClient;
+
+import java.util.Objects;
+
+/**
+ * Additional parameters to build {@link SearchField}.
+ */
+public class FieldBuilderOptions {
+    private MemberNameConverter serializer;
+
+    /**
+     * Gets the serializer use to build search fields in
+     * {@link SearchIndexClient#buildSearchField(Class, FieldBuilderOptions)} buildSearchField} or
+     * {@link SearchIndexAsyncClient#buildSearchField(Class, FieldBuilderOptions) buildSearchField}
+     *
+     * @return the custom serializer.
+     */
+    public MemberNameConverter getSerializer() {
```

Comment:
Do we want this to be `MemberNameConverter` or `JsonSerializer`? .NET is using `ObjectSerializer` as a parameter to `FieldBuilder`.

cc: @heaths @tg-msft @srnagar @JonathanGiles

Reply:
.NET also proposed to use the option class. The options class is a placeholder which can put more properties in future.

Reply:
Both `JsonSerializer` and `ObjectSerializer` miss the API of `convertMemberName`, which is the necessary one using in FieldBuilder.